src/content/11/en/part11d.md

```diff
@@ -125,6 +125,39 @@ For example: We have a project that uses hash based artifact builds for testing,
 
 In the above case, the software we release is tested because the CI system makes sure that tests are run on the code it is about to tag. It would not be incorrect to say that the project uses semantic versioning and simply ignore that the CI system tests individual developer branches/PRs with a hash based naming system. We do this because the version we care about (the one that is released) is given a semantic version.
 
+</div>
+
+<div class="tasks">
+
+### Exercise 11.5
+
+> TODO: make sure exercise number is correct
+
+Let's set up a workflow that will automatically increase (bump) the version when a pull request is merged into master and tag the release with the version number. We will use an open-source action developed by a third-party: `anothrNick/github-tag-action`. You can read the documentation for this action in its [README](https://github.com/anothrNick/github-tag-action).
+
+Start by creating the workflow the same way as in previous exercises. Set push to master as the trigger and use the same environment as in other actions. The workflow should have one job with two steps. The first step is the already familiar `actions/checkout@v2` as we need have the latest code in the build environment. The second step should be the tagging action itself:
+
+```yml
+    - name: Bump version and push tag
+      uses: anothrNick/github-tag-action@1.25.0
```

Comment:
I'm not sure if I mentioned this as a security issue but if I haven't we need to add it:

The @1.25.0 tag is really bad. Tags can be moved and malicious code can be inserted. It's ok for github's own actions (e.g. checkout) since we trust both github and their ability to keep their repo secure. 

For third party repos, the only safe way to refer to it is as a commit hash so, for example, that tag action should be referred to as:

anothrNick/github-tag-action@83fbc06cb8316c1bb147a682abe390a06787d252

The full SHA sum needs to be used. Not pretty but especially for actions that get access to sensitive information (e.g. your github token) this is required.

Reply:
@mhalmagiu Could you check out the updated version?

Reply:
GAH! I didn't realize that the UI had rendered the shortened commit hash because it was trying to be smart.

It should be:
`anothrNick/github-tag-action@83fbc06cb8316c1bb147a682abe390a06787d252`
and NOT
`anothrNick/github-tag-action@83fbc06`

The reason for this is because GitHub explicitly disallows tags that are alphanumeric and 32 characters long (like a commit hash) so that you can't make a tag that looks like a commit hash. This means that a full commit hash in the action 'name' guarantees the code being run.

It seems that your change has the partial hash (like github decided to display above). If we change that to the full hash, it's all good.

Reply:
Wow, I had no idea. Should we mention that here, or is it too much information? 🤣

Reply:
We could, but I think just using the correct hash in the example (and maybe saying that the full hash should be used) is enough.